src/dawn/Optimizer/PassDataLocalityMetric.cpp

```diff
@@ -286,6 +302,38 @@ computeReadWriteAccessesLowerBound(StencilInstantiation* instantiation,
 
 } // anonymous namespace
 
+/// @brief Approximate the reads and writes individually for each ID
+std::unordered_map<int, ReadWriteAccumulator>
+computeReadWriteAccessesMetricPerAccessID(StencilInstantiation* instantiation,
+                                          const MultiStage& multiStage,
+                                          const HardwareConfig& config) {
+  ReadWriteCounter readWriteCounter(instantiation, multiStage, config);
+
+  for(const auto& stage : multiStage.getStages())
+    for(const auto& DoMethod : stage->getDoMethods())
+      for(const auto& statementAccessesPair : DoMethod->getStatementAccessesPairs()) {
+        statementAccessesPair->getStatement()->ASTStmt->accept(readWriteCounter);
+      }
+
+  return readWriteCounter.getIndividualReadWrites();
+}
+
+
+/// @brief Approximate the reads and writes accoding to our data locality metric
+std::pair<int, int> computeReadWriteAccessesMetric(StencilInstantiation* instantiation,
+                                                   const MultiStage& multiStage,
+                                                   const HardwareConfig& config) {
+  ReadWriteCounter readWriteCounter(instantiation, multiStage, config);
+
+  for(const auto& stage : multiStage.getStages())
+    for(const auto& DoMethod : stage->getDoMethods())
```

Comment:
question about convention: do we have a convention for if, ...scopes? I would always  {} (as opposed to single line) since it can avoid mistakes when we add more statements. But if we have a convention for single line without {} already is fine.... as far as we are consistent

Reply:
I never use the {} for a single line (I think it is also stated by the LLVM coding style). Imho it just pollutes the code with unnecessary braces.